File: src/with_java/RomanToInt.java
class Solution {
    public int romanToInt(String s) {
        int len = s.length();
        int[] numbers = new int[len];
        for (int i = 0; i < len; i++) {
            numbers[i] = map(s.charAt(i));
        }
        int sum = 0;
        for (int i = len - 1; i >= 0; i--) {
            if (i == 0) {
                sum += numbers[i];
                break;
            }
            if (numbers[i] <= numbers[i - 1]) {
                sum += numbers[i];
            } else {
                sum = sum + numbers[i] - numbers[i - 1];
                i--;
            }
        }
        return sum;
    }
    
    public int map(char c) {
        switch(c) {
            case('V'):
                return 5;
            case('X'):
                return 10;
            case('L'):
                return 50;
            case('C'):
                return 100;
            case('D'):
                return 500;
            case('M'):
                return 1000;
            default:
                return 1;
        }
    }
}

File: src/with_kotlin/RomanToInt.kt
class Solution {
    fun romanToInt(s: String): Int {
        val len = s.length
        val numbers = Array<Int>(len) { 0 }
        for (i in 0 until len) {
            numbers[i] = map(s[i])
        }
        var sum = 0
        var j = len - 1
        do {
            if (j == 0) {
                sum += numbers[j]
                break
            }
            if (numbers[j] <= numbers[j - 1]) {
                sum += numbers[j]
            } else {
                sum = sum + numbers[j] - numbers[j - 1]
                j -= 1
            }
            j -= 1
        } while (j >= 0)
        return sum
    }
    
    fun map(s: Char) = when(s) {
        'V' -> 5
        'X' -> 10
        'L' -> 50
        'C' -> 100
        'D' -> 500
        'M' -> 1000
        else -> 1
    }
}